### PR TITLE
lib-schema JSDoc/TS fixes #11991

### PR DESCRIPTION
**Pull request by rymsha**
Part of #11991.

Audit findings for `lib-schema` — Java handlers in
`modules/lib/lib-schema/src/main/java/...` are the runtime truth, so
only the JSDoc/TS declarations in `schema.ts` were updated.

### Fixes

- **`StyleDescriptor`**: drop `cssPath` and `elements[].element` —
  `StyleDescriptorMapper` never emits either field. The
  `getStyles.js` example confirms the runtime shape.
- **`getSchema`**: handler returns `null` when the schema is not
  found; declared the return type and JSDoc as nullable.
- **`getSite`, `getStyles`**: TS already had `| null`; JSDoc updated
  to match.
- **`getComponent`**:
  - Internal `GetDynamicComponentHandler.execute()` was typed as
    `ContentTypeSchema | FormFragmentSchema | MixinSchema` (the
    schema union, copy/paste from `getSchema`); corrected to
    `LayoutDescriptor | PageDescriptor | PartDescriptor`.
  - JSDoc `@returns {string}` and function return `unknown` were
    both wrong; now reflect the descriptor union.
- **`createComponent`, `updateSite`, `updateStyles`**: JSDoc
  `@returns {string}` corrected to the actual descriptor types.
- **`create*` / `update*` params**: `resource` is required at the
  JS layer (each function calls `checkRequired(params, 'resource')`
  which throws when missing). JSDoc had `[params.resource]`
  (optional) and `CreateDynamicContentSchemaParams` typed it as
  `resource?: string | null`; both updated to required so JSDoc and
  TS match the user-facing contract.
- **`update*` JSDoc**: changed `@returns ... created resource` to
  `updated resource`.
- **`listSchemas`**: return type listed `ContentSchemaType[]` (which
  is `('CONTENT_TYPE' | 'FORM_FRAGMENT' | 'MIXIN')[]`) — clearly
  wrong; corrected to `ContentTypeSchema[]` in both TS and JSDoc.
- **`getStyles` JSDoc**: capitalize `Application key.` for
  consistency with sibling functions.

Java handler behavior is unchanged — only JSDoc and TS type
declarations were updated to match what the code does.

`./gradlew :lib:lib-schema:test` runs green locally
(`BUILD SUCCESSFUL`). No `integrationTest` source set in this
module.

### CODE REVIEW

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts [hunk 1]
```diff
@@ -329,7 +325,7 @@ interface GetDynamicSiteHandler {
  * @param {object} params JSON with the parameters.
  * @param {string} params.application Application key.
  *
- * @returns {SiteDescriptor} fetched resource.
+ * @returns {SiteDescriptor | null} fetched resource, or `null` if not found.
```
[the remark this answers is not in the document]
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts:334]: Skipped: this PR is a JSDoc/TS audit (#11991) that aligns declarations with the Java handlers — adding lib-level regression tests for null returns is out of scope. Filing a separate issue for the test gap is the right next step.

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts [hunk 2]
```diff
@@ -353,9 +349,9 @@ interface GetDynamicStylesHandler {
  * Fetches dynamic styles schema resource.
  *
  * @param {object} params JSON with the parameters.
- * @param {string} params.application application key.
+ * @param {string} params.application Application key.
  *
- * @returns {StyleDescriptor} fetched resource.
+ * @returns {StyleDescriptor | null} fetched resource, or `null` if not found.
```
[the remark this answers is not in the document]
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts:360]: Skipped: same reasoning as the getSite comment — adding lib-level tests stubbing the service to return null is a separate task from this JSDoc/TS audit.

**Comment by rymsha** [comment]
Reviewed Copilot feedback: applied 2, skipped 2 (replies inline).

Pushed `427ccb5db3` on top of the audit commit (squashed into a single commit per repo convention).

**Applied**
- `CreateDynamicContentSchemaParams.type` tightened from `string` to `ContentSchemaType` (and the matching `setType` on the handler interface), so it lines up with the rest of the schema APIs and the Java `DynamicContentSchemaType.valueOf(type)` call site.
- `StyleDescriptor.elements` made required and `elements[].label` made `string | null`, matching `StyleDescriptorMapper` (always emits the array, even empty) and `Style` (only `name` is validated as required; `label` is optional).

**Skipped (out of scope)**
- Adding lib-level regression tests for `getSite()` / `getStyles()` returning `null` — valuable, but this PR is a JSDoc/TS audit (#11991) and tests should be a separate change.

`./gradlew :lib:lib-schema:test` ran green locally after the changes (TS lib build + Java tests).

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts [hunk 3]
```diff
@@ -208,12 +206,10 @@ interface CreateDynamicStylesHandler {
...
     application: string;
-    cssPath: string;
     modifiedTime: string;
     resource: string;
-    elements?: {
-        element: string;
-        label: string;
+    elements: {
+        label: string | null;
         name: string;
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts:217]: What about `type`, `editor.css`?

Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts:217]: @copilot complete the implementation

File: modules/lib/lib-schema/src/main/java/com/enonic/xp/lib/schema/mapper/StyleDescriptorMapper.java [hunk 4]
```diff
@@ -5,12 +5,15 @@
...
 import com.enonic.xp.script.serializer.MapGenerator;
 import com.enonic.xp.script.serializer.MapSerializable;
+import com.enonic.xp.style.ImageStyle;
 import com.enonic.xp.style.Style;
 import com.enonic.xp.style.StyleDescriptor;
 
 public class StyleDescriptorMapper
     implements MapSerializable
 {
+    private static final String IMAGE_TYPE = "image";
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/java/com/enonic/xp/lib/schema/mapper/StyleDescriptorMapper.java]:
```suggestion
    private static final String IMAGE_TYPE = "Image";
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js [hunk 5]
```diff
@@ -52,11 +52,13 @@ assert.assertJsonEquals({
     elements: [
         {
             label: 'Override ${width}',
-            name: 'editor-width-auto'
+            name: 'editor-width-auto',
+            type: 'image'
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js]:
```suggestion
            type: 'Image'
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js [hunk 6]
```diff
@@ -52,11 +52,13 @@ assert.assertJsonEquals({
...
         },
         {
             label: 'Cinema',
-            name: 'editor-style-cinema'
+            name: 'editor-style-cinema',
+            type: 'image'
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js]:
```suggestion
            type: 'Image'
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/getStyles.js [hunk 7]
```diff
@@ -21,7 +21,8 @@ assert.assertJsonEquals({
     elements: [
         {
             label: 'Style display name',
-            name: 'mystyle'
+            name: 'mystyle',
+            type: 'image'
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/getStyles.js]:
```suggestion
            type: 'Image'
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js [hunk 8]
```diff
@@ -52,11 +52,13 @@ assert.assertJsonEquals({
     elements: [
         {
             label: 'Override ${width}',
-            name: 'editor-width-auto'
+            name: 'editor-width-auto',
+            type: 'image'
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js]:
```suggestion
            type: 'Image'
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js [hunk 9]
```diff
@@ -52,11 +52,13 @@ assert.assertJsonEquals({
...
         },
         {
             label: 'Cinema',
-            name: 'editor-style-cinema'
+            name: 'editor-style-cinema',
+            type: 'image'
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js]:
```suggestion
            type: 'Image'
```

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts [hunk 10]
```diff
@@ -208,13 +206,13 @@ interface CreateDynamicStylesHandler {
...
+        type: string;
+        editor?: Record<string, unknown>;
```
Comment (anatol-sialitski) [inline modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts]: Use  ConfigValue and ConfigObject from index.d.ts

export type ConfigValue = string | number | boolean | ConfigObject | ConfigValue[];

export interface ConfigObject {
    [key: string]: ConfigValue
}

then

type EditorConfig = {
  css?: string;
  [key: string]: ConfigValue;
};

editor?: EditorConfig;

File: modules/lib/lib-schema/src/main/java/com/enonic/xp/lib/schema/mapper/StyleDescriptorMapper.java (hunk 4 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/java/com/enonic/xp/lib/schema/mapper/StyleDescriptorMapper.java]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js (hunk 5 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js (hunk 6 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/createStyles.js]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/getStyles.js (hunk 7 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/getStyles.js]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js (hunk 8 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js (hunk 9 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/examples/schema/updateStyles.js]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c.

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts (hunk 10 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts]: Applied in 58bf79114fc7940d0316d75e2879f2e2af16775c — added `export type EditorConfig = { css?: string; [key: string]: ConfigValue }` and switched the `editor?` field to use it. `ConfigValue` was already imported from `@enonic-types/core`.

**Review by anatol-sialitski — APPROVED** [review]

File: modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts (hunk 10 above)
Comment (rymsha) [reply modules/lib/lib-schema/src/main/resources/lib/xp/schema.ts]: Heads-up: the literal form `{ css?: string; [key: string]: ConfigValue }` doesn't typecheck (TS2411 — `string | undefined` is not assignable to `ConfigValue`), so 58bf791 broke the `:lib:typescript` task. Fixed in b8ea608 by expressing the same intent as `ConfigObject & { css?: string }`.

**Comment by rymsha** [comment]
Follow-up #12019 surfaces `aspectRatio` and `filter` on `ImageStyle` through the lib-schema JS API — the underlying parser already bound both fields, only `StyleDescriptorMapper` was dropping them. Opening as a separate PR rather than reopening this branch (matches the pattern used by #12018).

**Comment by rymsha** [comment]
Confirmed: the `:lib:typescript` task runs clean on head commit [`b8ea608`](https://github.com/enonic/xp/commit/b8ea608b1adb6395a3a8e3d94646d1aab7f76628) — `EditorConfig = ConfigObject & { css?: string }` resolves the TS2411 collision. Build [run](https://github.com/enonic/xp/actions/runs/25315140170/job/74210891590) green.
